src/data/repositories/MetadataD2ApiRepository.ts

```diff
@@ -95,7 +95,79 @@ export class MetadataD2ApiRepository implements MetadataRepository {
     }
 
     private fetchMetadataWithDependencies(model: MetadataModel, id: string): FutureData<MetadataPayload> {
-        return apiToFuture<MetadataPayload>(this.api.get(`/${model}/${id}/metadata.json`));
+        switch (model) {
+            case "dataElementGroupSets":
+                return this.getDataElementGroupSetWithDependencies(id);
+            case "organisationUnitGroups":
+                return this.getOrgUnitGroupDependencies(id);
+            case "organisationUnitGroupSets":
+                return this.getOrgUnitGroupSetWithDependencies(id);
+            default:
+                return apiToFuture<MetadataPayload>(this.api.get(`/${model}/${id}/metadata.json`));
```

Comment:
Just a note to keep in mind when we add support for other metadata:

[According the docs the /metadata.json](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-241/metadata.html#webapi_dataset_program_export_dependencies) endpoint is only available for these models:

```
/api/dataSets/{id}/metadata.json

/api/programs/{id}/metadata.json

/api/categoryCombos/{id}/metadata.json

/api/dashboards/{id}/metadata.json

/api/optionSets/{id}/metadata.json

/api/dataElementGroups/{id}/metadata.json
```